inc/features/blocks/namespace.php

```diff
@@ -69,3 +78,248 @@ function get_block_settings( string $name ) : ?array {
+ * @return bool
+ */
+function validate_id( $param ) : bool {
+	return (bool) preg_match( '/[a-z0-9-]+/', $param );
+}
+
+/**
+ * Sanitize Experience Block ID.
+ *
+ * @param string $param The Experience Block ID.
+ * @return string
+ */
+function sanitize_id( $param ) : string {
+	return preg_replace( '/[^a-z0-9-]+/', '', $param );
+}
+
+/**
+ * Check current user can view XB analytics data.
+ *
+ * @return boolean
+ */
+function check_views_permission() : bool {
+	return true;
+	$type = get_post_type_object( Audiences\POST_TYPE );
+	return current_user_can( $type->cap->edit_posts );
+}
+
+/**
+ * Retrieve the Experience Block views data.
+ *
+ * @param WP_REST_Request $request The REST request object.
+ * @return WP_REST_Response
+ */
+function handle_views_request( WP_REST_Request $request ) : WP_REST_Response {
+	$block_id = $request->get_param( 'id' );
+	$post_id = $request->get_param( 'post_id' ) ?? null;
+	$views = get_views( $block_id, $post_id );
+	return rest_ensure_response( $views );
+}
+
+/**
+ * Get the Experience Block views data.
+ *
+ * @param string $block_id The Experience block client ID to get data for.
+ * @param int|null $post_id An optional post ID to limit results by.
+ * @return array|WP_Error
+ */
+function get_views( string $block_id, ?int $post_id = null ) {
+	$query = [
+		'query' => [
+			'bool' => [
+				'filter' => [
+					// Set current site.
+					[
+						'term' => [
+							'attributes.blogId.keyword' => get_current_blog_id(),
+						],
+					],
+
+					// Set block ID.
+					[
+						'term' => [
+							'attributes.clientId.keyword' => $block_id,
+						],
+					],
+
+					// Last month.
+					[
+						'range' => [
+							'event_timestamp' => [
+								'gte' => Utils\milliseconds() - ( MONTH_IN_SECONDS * 1000 ),
+							],
+						],
+					],
+
+					// Limit event type to experienceView.
+					[
+						'term' => [
+							'event_type.keyword' => 'experienceView',
+						],
+					],
+				],
+			],
+		],
+		'aggs' => [
+			// Get the split by audience.
+			'audiences' => [
+				'terms' => [
+					'field' => 'attributes.audience.keyword',
+				],
+			],
+
+			// Get the split by post ID and audience.
+			'posts' => [
+				'terms' => [
+					'field' => 'attributes.postId.keyword',
+				],
+				'aggs' => [
+					'audiences' => [
+						'terms' => [
+							'field' => 'attributes.audience.keyword',
+						],
+					],
+				],
+			],
+		],
+		'size' => 0,
+		'sort' => [
+			'event_timestamp' => 'desc',
+		],
+	];
+
+	// Add post ID query filter.
+	if ( $post_id ) {
+		// Remove the posts aggregation.
+		unset( $query['aggs']['posts'] );
+
+		$query['query']['bool']['filter'][] = [
+			'term' => [
+				'attributes.postId.keyword' => (string) $post_id,
+			],
+		];
+	}
+
+	$key = sprintf( 'views:%s', "{$block_id}:{$post_id}" );
+	$cache = wp_cache_get( $key, 'altis-xbs' );
+	if ( $cache ) {
+		return $cache;
+	}
+
+	$result = Utils\query( $query );
+
+	if ( ! $result ) {
```

Comment:
Should no data be cached too?

Reply:
This is more a failure scenario rather than no data - no data will report successfully but with zero hits.